algorithms/arrays/longest_non_repeat.py

```diff
@@ -21,7 +21,7 @@ def longest_non_repeat_v1(string):
     max_len = 0
     for i in string:
         if i in temp:
-            temp = []
+            temp = temp[temp.index(i) + 1:]
```

Comment:
@newbility77 Nice catch! We could further optimize this code. Now it requires 2 traversals of `temp` in line`23` and `24`.
We can do in 1 traversal:
```py
for i in string:
    for idx, c in enumerate(temp):
        if c == i:
            temp = temp[idx + 1:]
            break
    temp.append(i)
    ...
```